File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/core/JavaScriptFunctionCallback.kt
package ca.logaritm.dezel.core

/**
 * @class JavaScriptFunctionCallback
 * @super JavascriptCallback
 * @since 0.1.0
 */
public class JavaScriptFunctionCallback(context: JavaScriptContext, target: Long, callee: Long, argc: Int, argv: LongArray) : JavaScriptCallback(context, target, callee, argc, argv) {

	//--------------------------------------------------------------------------
	// Methods
	//--------------------------------------------------------------------------

	/**
	 * @method argument
	 * @since 0.1.0
	 */
	public fun argument(index: Int, protect: Boolean = true): JavaScriptValue {
		return if (index < this.argc) JavaScriptValue.create(this.context, this.argv[index], bridge = true) else this.context.Undefined
	}
}

File: platforms/ios/Dezel/Dezel/Core/JavaScriptFunctionCallback.swift
/**
 * The arguments object provided to a function.
 * @class JavaScriptFunctionCallback
 * @since 0.1.0
 */
public final class JavaScriptFunctionCallback: JavaScriptCallback {

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * Returns an argument at a specified index.
	 * @method argument
	 * @since 0.1.0
	 */
	final public func argument(_ index: Int) -> JavaScriptValue {
		return JavaScriptValue.create(self.context, handle: self.argv[index]!)
	}

	/**
	 * Returns an argument at a specified index.
	 * @method argument
	 * @since 0.1.0
	 */
	final public func argument(_ index: Int, protect: Bool) -> JavaScriptValue {
		return JavaScriptValue.create(self.context, handle: self.argv[index]!, protect: protect)
	}
}
